policies/policies.go

```diff
@@ -29,26 +29,92 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"os"
+	"strings"
 
 	"clouditor.io/clouditor/api/evidence"
 	"github.com/open-policy-agent/opa/rego"
 )
 
-func RunEvidence(file string, evidence *evidence.Evidence) (data map[string]interface{}, err error) {
+// applicableMetrics stores a list of applicable metrics per resourceType
+var applicableMetrics = make(map[string][]string)
+
+func RunEvidence(evidence *evidence.Evidence) ([]map[string]interface{}, error) {
+	data := make([]map[string]interface{}, 0)
+	var baseDir string = "."
+	// check, if we are in the root of Clouditor
+	if _, err := os.Stat("policies"); os.IsNotExist(err) {
+		// in tests, we are relative to our current package
+		baseDir = ".."
+	}
+
 	var m = evidence.Resource.GetStructValue().AsMap()
 
-	return RunMap(file, m)
+	var types []string
+
+	if rawTypes, ok := m["type"].([]interface{}); ok {
+		types = make([]string, len(rawTypes))
+	} else {
+		return nil, fmt.Errorf("got type '%T' but wanted '[]interface {}'. Check if resource types are specified ", rawTypes)
+	}
+	for i, v := range m["type"].([]interface{}) {
+		// TODO(all): type assertion check good or unnecessary because we assume resourceTypes to be always set as intended ([]string)?
+		if t, ok := v.(string); !ok {
+			return nil, fmt.Errorf("got type '%T' but wanted 'string'", t)
+		} else {
+			types[i] = t
+		}
+	}
+	// TODO(lebogg): Try to optimize duplicated code
+	if key := createKey(types); applicableMetrics[key] == nil {
+		files, err := scanBundleDir(baseDir)
+		if err != nil {
+			return nil, fmt.Errorf("could not load metric bundles: %v", err)
+		}
+
+		for _, fileInfo := range files {
+			//file := fmt.Sprintf("%s/policies/policyBundles/%s/", baseDir, fileInfo.Name())
+			runMap, err := RunMap(baseDir, fileInfo.Name(), m)
+			if err != nil {
+				return nil, err
+			}
+			if runMap != nil {
+				applicableMetrics[key] = append(applicableMetrics[key], fileInfo.Name())
+				data = append(data, runMap)
+			}
+		}
+	} else {
+		for _, metric := range applicableMetrics[key] {
+			//bundle := fmt.Sprintf("%s/policies/policyBundles/%s/", baseDir, metric)
+			runMap, err := RunMap(baseDir, metric, m)
+			if err != nil {
+				return nil, err
+			}
+			data = append(data, runMap)
+		}
+	}
+	return data, nil
 }
 
-func RunMap(file string, m map[string]interface{}) (data map[string]interface{}, err error) {
+func RunMap(baseDir string, metric string, m map[string]interface{}) (data map[string]interface{}, err error) {
 	var (
 		ok bool
 	)
 
+	// Create paths for bundle directory and utility functions file
+	bundle := fmt.Sprintf("%s/policies/policyBundles/%s/", baseDir, metric)
```

Comment:
maybe just use `/policies/`? I would avoid the camelCase directory names

Reply:
I would suggest `bundles` instead of `policyBundles`. I do not want to blow up the policies directory where also, e.g., the `policies.go` file lies

Reply:
Ok